Xamarin.Forms.Core/IndicatorView.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Xamarin.Flex;
+using Xamarin.Forms.Platform;
+
+namespace Xamarin.Forms
+{
+	[ContentProperty(nameof(IndicatorLayout))]
+	[RenderWith(typeof(_IndicatorViewRenderer))]
+	public class IndicatorView : TemplatedView
+	{
+		public static readonly BindableProperty PositionProperty = BindableProperty.Create(nameof(Position), typeof(int), typeof(IndicatorView), default(int), BindingMode.TwoWay, propertyChanged: (bindable, oldValue, newValue)
+			=> ((IndicatorView)bindable).ResetIndicatorStyles());
+
+		public static readonly BindableProperty CountProperty = BindableProperty.Create(nameof(Count), typeof(int), typeof(IndicatorView), default(int), propertyChanged: (bindable, oldValue, newValue)
+			=> ((IndicatorView)bindable).ResetIndicatorCount((int)oldValue));
+
+		public static readonly BindableProperty MaximumVisibleCountProperty = BindableProperty.Create(nameof(MaximumVisibleCount), typeof(int), typeof(IndicatorView), int.MaxValue, propertyChanged: (bindable, oldValue, newValue)
+			=> ((IndicatorView)bindable).ResetIndicatorStyles());
+
+		public static readonly BindableProperty IndicatorTemplateProperty = BindableProperty.Create(nameof(IndicatorTemplate), typeof(DataTemplate), typeof(IndicatorView), default(DataTemplate), propertyChanged: (bindable, oldValue, newValue)
```

Comment:
accepting templates is allowing the user to pass a full view as the mini, and that will blow up, perf-wise.

instead of template, color, selected color (btw, it's quite hard to set the color on a templated item), I would take Image, color, selectedcolor, and tint the image according to the state. (By tinting, I mean desaturating the image, and tinting it using the color)